web/packages/teleterm/src/ui/ConnectMyComputer/NavigationMenu.tsx

```diff
@@ -21,39 +21,62 @@ import { Box } from 'design';
 
 import { useAppContext } from 'teleterm/ui/appContextProvider';
 import { ListItem } from 'teleterm/ui/components/ListItem';
+import { ClusterUri } from 'teleterm/ui/uri';
 
-import { ConnectMyComputerIcon } from './ConnectMyComputerIcon';
+import { NavigationMenuIcon } from './NavigationMenuIcon';
+import { canUseConnectMyComputer } from './permissions';
 
-export function ConnectMyComputer() {
+interface NavigationMenuProps {
+  clusterUri: ClusterUri;
+}
+
+export function NavigationMenu(props: NavigationMenuProps) {
   const iconRef = useRef();
   const [isPopoverOpened, setIsPopoverOpened] = useState(false);
   const appCtx = useAppContext();
+  const cluster = appCtx.clustersService.findCluster(props.clusterUri);
```

Comment:
If we're going to return early if `cluster.leaf`, wouldn't it require less logic to  return early after the first hook and then change this line to:

 ```suggestion
  const rootCluster = appCtx.clustersService.findCluster(props.clusterUri);
```

Otherwise we end up with two variables storing the same information but obtained in a different way.

Alternatively, we could do something like this:

```typescript
  const cluster = appCtx.clustersService.findCluster(props.clusterUri);
  // …
  if (cluster.leaf) {
    return null;
  }

  const rootCluster = cluster;
```

Reply:
I like the alternative solution, it's much simpler.